### ISSUE TITLE
Best results on old / simple hardware?

### ISSUE THREAD

**Issue by pete-789** [ultralytics/ultralytics#9910]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

I'm after some broad-brush advice on how to get the best speed out of the hardware and software I'm using. Hopefully I can drill down to work out the detail myself.

I have been running a slightly modified version of the Speed Estimation routine (Solutions folder, YOLOv8) on a fairly old Windows machine. It runs well but the fps has been poor with a USB camera and no GPU to speed things up.

So I've bought a J1020 (basically a Jetson Nano on a carrier board, I know, it's old, old, old...), and a CSI camera (a Raspberry Pi v2).

I will be loading a pre-made OS image from https://github.com/Qengineering/Jetson-Nano-Ubuntu-20-image (Ubuntu 20.04 with OpenCV, TensorFlow, Pytorch and TensorFlowRT) and installing YOLOv8 (I can use an M2 256GB SSD drive, so have loads of room).

My requirements are to detect and track motor vehicles ONLY - cars, trucks, buses, motorcycles - hopefully in real time, from about 20m away. The viewpoint will always be at 90 degrees to the road (at right angles in other words), so shapes should be simple and fairly consistent.

This thesis https://medium.com/@sktedro/yolov8-benchmark-for-vehicle-detection-4df2d28e360a suggests that frame rates can be quite fast even on a Nano if the image size is reduced (his 'Pico' and 'Femto' styled models).

My question is - can I just reduce the image size of the stream being fed to YOLOv8? Or do I have to train based on the smaller image sizes ?  Apologies for my lack of knowledge...

Any other tips or pointers for achieving the best speed would be gratefully received !  I love how easy YOLOv8 is to use BTW.

**Comment by glenn-jocher** [comment]
Hello! It's great to hear that you're exploring YOLOv8 for vehicle detection on your J1020 setup. 🚗

To address your question about reducing the image size: yes, reducing the image size can indeed help increase the frame rate, especially on hardware with limited resources like the Jetson Nano. You can adjust the `imgsz` parameter when running predictions to reduce the input image size. Here’s a quick example:

```bash
yolo detect predict model=yolov8n.pt source='camera_stream' imgsz=320
```

This command sets the image size to 320x320, which should be lighter on your hardware.

You don't necessarily need to retrain your model on smaller image sizes unless you find that the detection accuracy significantly drops at the reduced resolution you're using for inference.

Additionally, make sure to:
- Enable maximum power mode (`sudo nvpmodel -m 0` and `sudo jetson_clocks`) on your Jetson to ensure it runs at peak performance.
- Consider using a lighter model variant like YOLOv8n if you're not already doing so, as it's designed to be faster and more efficient on less powerful devices.

Hope this helps, and enjoy your vehicle detection project with YOLOv8! 🚀

**Comment by pete-789** [comment]
That's great to know, thanks for the reply!

**Comment by glenn-jocher** [comment]
You're welcome! If you have any more questions or need further assistance, feel free to ask. Happy coding! 😊
